src/App.js

```diff
@@ -1,16 +1,48 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.js';
+// import InfoBar from './components/InfoBar.js';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+  const [hearts, setHearts] = useState(0);
+
+  const updateLikes = (id) => {
+    const updatedEntries = entries.map((entry) => {
+      if (entry.id === id) {
+        const updatedEntry = {...entry};
+        updatedEntry.liked = !updatedEntry.liked;
+        if (updatedEntry.liked) {
+          setHearts(hearts+1);
+        } else {
+          setHearts(hearts-1);
+        }
```

Comment:
Love that you update the state of the `hearts` in this larger `updateLikes()` function!